toqito/matrices/comparison.py

```diff
@@ -0,0 +1,49 @@
+"""Computes comparison matrix."""
+
+import numpy as np
+
+
+def comparison(mat: np.ndarray) -> np.ndarray:
+    r"""Compute comparison matrix of a given square matrix.
+
+    This function computes the comparison matrix :math:`M(A)` for a square matrix :math:`A` as defined in
+    :cite:`WikiComparisonMatrix`. For each entry, the diagonal entries are given by the absolute value of the original
+    diagonal entries of :math:`A`, while the off-diagonal entries are given by minus the absolute value of the
+    corresponding entries. In other words,
+
+    .. math::
+        m_{ij} =
+        \begin{cases}
+        |a_{ij}|, & \text{if } i = j, \\
+        -|a_{ij}|, & \text{if } i \neq j.
+        \end{cases}
+
+    Examples
+    ========
+    >>> import numpy as np
+    >>> from matrices.comparison import comparison_matrix
+    >>> A = np.array([[2, -1],
+    ...               [3, 4]])
+    >>> comparison_matrix(A)
+    array([[ 2., -1.],
+        [-3.,  4.]])
+
+    References
+    ==========
+    .. bibliography::
+        :filter: docname in docnames
+
+
+    :param mat: The input square matrix.
+    :raises ValueError: If the input matrix is not square.
+    :return: The comparison matrix of the input matrix.
+
+    """
+    if len(mat.shape) != 2 or mat.shape[0] != mat.shape[1]:
+        raise ValueError("Input matrix must be square.")
```

Comment:
Can't you use `is_square` here?

Reply:
Unfortunately, if we do, it causes a circular import error so I opted for checking explicitly. Related: https://github.com/vprusso/toqito/issues/1081

Reply:
I really hope this issue gets picked up by someone to work on.

Reply:
You and me both!